js/main.js

```diff
@@ -28,26 +29,31 @@ function getCSVData(callback = undefined) {
   });
 }
 
+function getCardsByRarity() {
+  cards_by_rarity = cards_data.reduce((result, card) => {
+    result[card["Rarity Folder"]] = [
+      ...(result[card["Rarity Folder"]] || []),
+      card,
+    ];
+    return result;
+  }, {});
+}
+
 async function main() {
   await defineCardComponent();
-  getCSVData((cards_data) => {
-    if (CURRENT_PAGE == "/collection.html") {
-      renderCards(cards_data, COLLECTIONS_MAIN_CONTENT);
+  getCSVData(async (cards_data) => {
+    switch (CURRENT_PAGE) {
+      case "/gacha.html":
+        GACHA_BUTTON.onclick = (event) =>
+          pullAndRenderCards(cards_data, COLLECTIONS_MAIN_CONTENT);
+        await setupDetailsDialog();
+        break;
+      case "/collection.html":
+        await setupDetailsDialog();
+        renderCards(cards_data, COLLECTIONS_MAIN_CONTENT);
     }
+    getCardsByRarity();
```

Comment:
(nit) `getCardsByRarity` feels more intuitive to me to be next to the cards_data setup in getCSVData. Move it there?

Reply:
Moved there o7